appium/lib/screen-object.rb

```diff
@@ -66,82 +69,136 @@ def wait_step(timeout = 5, message = nil, &block)
-    for i in 0..num_loop
-      begin
-        if (driver.find_element(locator,locator_value).displayed?)
-          driver.find_element(locator,locator_value).click
-          break
-        end
-      rescue
-        driver.swipe(:start_x => 0,:start_y => scrollStart,:end_x =>0,:end_y =>scrollEnd,:touchCount => 2,:duration => 0)
-        false
-      end
+  # Find the first element containing value
+  # @param value [String] the value to search for
+  # @return [Element]
+  def get_element_by_text(value)
+    if value.to_s.strip.empty?
+      raise('parameter for get_element_by_text function cannot be empty string')
+    else
+      driver.find(value)
     end
   end
 
-  def scroll_up_find(locator,locator_value,num_loop = 15)
-    scr = driver.window_size
-    screenHeightStart = (scr.height) * 0.5
-    scrollStart = screenHeightStart.to_i
-    screenHeightEnd = (scr.height) * 0.2
-    scrollEnd = screenHeightEnd.to_i
-    for i in 0..num_loop
-      begin
-        if (driver.find_element(locator,locator_value).displayed?)
-          break
-        end
-      rescue
-        driver.swipe(:start_x => 0,:start_y => scrollEnd,:end_x =>0,:end_y =>scrollStart,:touchCount => 2,:duration => 0)
-        false
-      end
+  # Find the first element exactly matching value
+  # @param value [String] the value to search for
+  # @return [Element]
+  def get_element_by_exact_text(value)
+    if value.to_s.strip.empty?
+      raise('parameter for get_element_by_exact_text function cannot be empty string')
+    else
+      driver.find_exact(value)
     end
   end
 
+  # Create an object to exactly match the first element with target value
+  # @param value [String] the value to search for
+  # @return [String]
+  def webview_text_visible?(value)
+    driver.string_visible_exact('*', value)
+  rescue RuntimeError => e
+    raise("Could not find text \"#{value}\" on the current screen: #{e}")
+  end
 
-  def scroll_up_click(locator,locator_value,num_loop = 15)
-    scr = driver.window_size
-    screenHeightStart = (scr.height) * 0.5
-    scrollStart = screenHeightStart.to_i
-    screenHeightEnd = (scr.height) * 0.2
-    scrollEnd = screenHeightEnd.to_i
-    for i in 0..num_loop
-      begin
-        if (driver.find_element(locator,locator_value).displayed?)
-          driver.find_element(locator,locator_value).click
-          break
-        end
-      rescue
-        driver.swipe(:start_x => 0,:start_y => scrollEnd,:end_x =>0,:end_y =>scrollStart,:touchCount => 2,:duration => 0)
-        false
-      end
-    end
+  # Click on the first element with target value that contains search value
+  # @param text [String] the value to search for
+  # @return [Nil]
+  def click_text(text)
+    get_element_by_text(text).click
+  end
+
+  # Click exact text that matches the first element with target value
+  # @param text [String] the value to search for
+  # @return [Nil]
+  def click_exact_text(text)
+    get_element_by_exact_text(text).click
```

Comment:
After moving `get_element_by_exact_text` to `element.rb`, update this reference to `ScreenObject::Accessors::Element.get_element_by_exact_text`